mlflow/projects/__init__.py

```diff
@@ -651,6 +653,18 @@ def _get_docker_command(image, active_run):
     cmd += tracking_cmds + artifact_cmds
     env_vars.update(tracking_envs)
     env_vars.update(artifact_envs)
+    if user_env_vars is not None:
+        for user_entry in user_env_vars:
+            if isinstance(user_entry, list):
+                # User has defined a new environment variable for the docker environment
+                env_vars[user_entry[0]] = user_entry[1]
+            else:
+                # User wants to copy an environment variable from system environment
+                env_vars[user_entry] = os.environ[user_entry]
```

Comment:
```suggestion
                system_var = os.environ.get(user_entry)
                if system_var is None:
                    raise MlflowException("This project expects the %s environment variables to be set on the machine running the project, but %s was not set. Please ensure all expected environment variables are set" % (", ".join(user_env_vars), user_entry))
```

The idea is basically to throw an informative error message telling the user which environment variables must be set in order to run the project